cpp/src/arrow/table.cc

```diff
@@ -522,61 +522,59 @@ Status Table::RenameColumns(const std::vector<std::string>& names,
   return Status::OK();
 }
 
-Status ConcatenateTables(const std::vector<std::shared_ptr<Table>>& tables,
-                         std::shared_ptr<Table>* table) {
+Result<std::shared_ptr<Table>> ConcatenateTables(
+    const std::vector<std::shared_ptr<Table>>& tables,
+    const ConcatenateTablesOptions options, MemoryPool* memory_pool) {
```

Comment:
This `const` doesn't seem to be here in `table.h`

Reply:
did you mean the const before ConcatenateTablesOptions?
it's intended as marking a value type (as oppose to reference / pointer) as const has not effect in a function declaration.